### PR TITLE
[BIOMAGE-1849] - Return rendering error

### PR DESCRIPTION
**Pull request by aerlaut**
# Description
#### Background
Errors are repeatedly reported because the current implementation of ErrorBoundary renders the bugged page. We want the error to be rendered and reported only when the UI can not render anymore (showing the "it's not you it's us") page.

#### Solution
Remove the `ErrorBoundary` component and use Next's built-in error handling mechanism. Move the Slack reporting functionality to the built-in mechanism.

### Testing
To test the changes, we'll be using an issue reported in this ticket: https://biomage.atlassian.net/browse/BIOMAGE-1820. You can use a new project or an existing one.

**Using a new project**
1. Create a new project
2. Upload some sample
3. Run data processing. Wait until the data is processed. Take note of the data processing URL.
4. Go back to Data Management.
5. Upload a new sample to the project.
6. DO NOT RUN DATA PROCESSING. Instead, visit the URL directly.

**Using existing project**
1. Visit Data Processing. Take note of the URL.
2. Go back to Data Management.
3. Upload a new sample to the project.
4. DO NOT RUN DATA PROCESSING. Instead, visit the URL directly.

You will see that the "It's not you it's us" error page will be shown. A new error log will be reported in the #error-logs channel. You can also trigger this error in ui-default and compare.

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1849

#### Link to staging deployment URL (or set N/A)
https://ui-agi-grumpy-baboon.scp-staging.biomage.net/

#### Links to any PRs or resources related to this PR
N/A

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [x] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: src/pages/_error.jsx [hunk 1]
```diff
@@ -1,73 +1,86 @@
...
+  const reduxState = useSelector((state) => state);
 
-        <Space direction='vertical' style={{ width: '100%' }}>
-          <Text>
-            If you need immediate help, or if the problem persists,
-            please leave feedback using the button below.
-            <br />
-            Thank you for your patience, we&apos;ll be up and running shortly.
-          </Text>
+  if (errorObject && environment === 'production') postErrorToSlack(errorObject, reduxState);
```
Comment (ivababukova) [inline src/pages/_error.jsx]: could you enable staging too?

File: src/utils/postErrorToSlack.js [hunk 2]
```diff
@@ -109,7 +102,7 @@ const postError = async (errorLog, context) => {
   }
 };
 
-const postErrorToSlack = async (error, info, reduxDump) => {
+const postErrorToSlack = async (error, reduxDump) => {
```
Comment (ivababukova) [inline src/utils/postErrorToSlack.js]: rename `reduxDump` to `reduxState` and `error` to `errorObject` to be consistent with the names used for these everywhere else.

**Review by ivababukova — COMMENTED** [review]
Code looks good. Before approving and merging, let's enable this for staging and test in staging to make sure that the behaviour is as intended.

File: src/utils/postErrorToSlack.js (hunk 2 above)
Comment (aerlaut) [reply src/utils/postErrorToSlack.js]: renamed

**Review by ivababukova — APPROVED** [review]

**Pull request merged**